### DISCUSSION TITLE
Possibility to automaticly assign stock to sales order & ship it

### DISCUSSION THREAD

**Discussion by AkitidaKamaki** [inventree/InvenTree#8282] [Q&A]
My use case for InvenTree is to keep track of internal stock of machine parts, and when used they go out of the inventory. But that means I have to assign every part from the inventory for every salesorder and then also make a shipment for it. Is there a way to automate the process of assigning and shipping?

**Comment by SchrodingersGat** [comment]
It depends on how you want to automate it. You need to be able to track which items go to which customer, etc, so there does need to be some user intervention / business logic.

You can always write a custom plugin to achieve this, but it depends on your requirements. Perhaps you could share some more context around what exactly you are trying to achieve?

**Reply by AkitidaKamaki** [reply]
The current process to add a purchase order is
Create -> add every item -> allocate every item from stock to shipment-> ship items -> mark as shipped
Manual->Manual->Automated->automated->manual
in that order I want those things automated. I can make a plugin for that if needed, is there any documentation about creating a custom plugin?
So that I only need to add items to the order and then when i press a button the order is automaticly fulfilled

**Comment by SchrodingersGat** [comment]
You can read the documentation here for creating a custom plugin: https://docs.inventree.org/en/stable/extend/plugins/

Regarding your requested automations - I can see a potential workflow addition where the user can select "auto allocate stock" when creating a new line item to the sales order - much as the "auto allocate" button works currently for build orders
